design-documents/testing/functional/test-package.md

```diff
@@ -0,0 +1,109 @@
+
+>There are couple reasons to separate tests from core distribution:
+>
+>Security:
+> - reduce possible attack surface (less code)
+> - the test code is not written with security in mind
+> - even if we protect the access via .htaccess, some sites run Nginx or skip .htaccess making this code accessible
+> - we had security issues in the past that resulted in remote code exec using code related to tests
+>
+>Future-proof:
+> - ability to release separately, on separate cadence
+> - matches the general plan for all other non-critical modules - we want to modularize and release separately things like payment extensions, shipping and more.
+
+
+### Proposed solution
+In order to be able to package Functional Tests separately and match [Module separation naming](https://github.com/magento/architecture/blob/master/design-documents/module-separation-naming.md)
+proposal this proposal goes to direction where for each module which can be covered with functional tests we will have sibling module with suffix `{ModuleName}FunctionalTest`.
+
+For example:
+ - `CatalogAdminUi` module will have sibling module `CatalogAdminUiFunctionalTest`
+ - `CatalogAdminUiFunctionalTest` will be dedicated only to have Magento Automated Functional Tests for `CatalogAdminUi` module
+ 
+This module will have his own structure according to MFTF requirements:
+```   
+   `Magento`
+       `CatalogAdminUiFunctionalTest`
+           `ActionGroup`
+           `Data`
+           `Metadata`
+           `Page`
+           `Section`
+           `Suite`
+           `Test`
+           `LICENSE.txt`
+           `README.md`
+           `composer.json`
+```
+
+`composer.json` - will have his own dependencies on other test modules and in `suggest` section it will have information about what Magento Module this test package covers.
+
+For example:
+```
+{
+    "name": "magento/module-catalog-admin-ui-functional-test",
+    "description": "N/A",
+    "config": {
+        "sort-packages": true
+    },
+    "require": {
+        "magento/magento2-functional-testing-framework": "*",
+        "magento/module-backend-functional-test": "*",
+        "magento/module-catalog-inventory-functional-test": "*",
+        "magento/module-store-functional-test": "*",
+        "magento/module-ui-functional-test": "*"
+    },
+    "suggest": {
+        "magento/module-catalog-admin-ui": "Test coverage for magento/module-catalog-admin-ui: ~100.0.0"
+    },
+    "type": "magento2-functional-test-module",
+    "license": [
+        "OSL-3.0",
+        "AFL-3.0"
+    ]
+}
+```
+
+#### Magento metapackage
+In Magento `magento/project-community-edition` we will be able to specify 2 metapackage under different blocks.
+Magento application will go to `require` section as it is right now and we will be able to put `magento/functional-test-project-community-edition` under `require-dev` section.
```

Comment:
We might want to move "magento/magento2-functional-testing-framework" from 'require-dev' to new meta package with tests as well